src/components/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,331 @@
+    defaultWeekStart: 1,
+  };
+
+  if (selectedDate) {
+    calendarOptions.defaultDate = selectedDate;
+  }
+  const { body, headers, month, navigation, year } = useCalendar(calendarOptions);
+
+  const handleNextClick = (): void => {
+    navigation.toNext();
+  };
+
+  const handlePreviousClick = (): void => {
+    navigation.toPrev();
+  };
+
+  const headerDate = new Date();
+  headerDate.setMonth(month);
+  headerDate.setFullYear(year);
+
+  return (
+    <DatePickerContainer
+      data-testid="datepicker-calendar-container"
+      isResponsive={false}
+      fillWidth={false}
+      orientation="vertical"
+      padding="sm"
+    >
+      <Container
+        isResponsive={false}
+        justifyContent="space-between"
+        orientation="horizontal"
+      >
+        <IconButton
+          icon="chevron-left"
+          onClick={handlePreviousClick}
+          size="sm"
+          type="ghost"
+        />
+        <UnselectableTitle>{headerDateFormatter.format(headerDate)}</UnselectableTitle>
+        <IconButton
+          icon="chevron-right"
+          onClick={handleNextClick}
+          size="sm"
+          type="ghost"
+        />
+      </Container>
+      <DateTable>
+        <thead>
+          <tr>
+            {headers.weekDays.map(({ key, value: date }) => {
+              return (
+                <DateTableHeader key={key}>
+                  {weekdayFormatter.format(date)}
+                </DateTableHeader>
+              );
+            })}
+          </tr>
+        </thead>
+        <tbody>
+          {body.value.map(({ key: weekKey, value: week }) => {
+            return (
+              <tr key={weekKey}>
+                {week.map(({ date, isCurrentMonth, key: dayKey, value: fullDate }) => {
+                  const isSelected = selectedDate
+                    ? isSameDate(selectedDate, fullDate)
+                    : false;
+                  const isCurrentDate = isSameDate(new Date(), fullDate);
+                  const isDisabled = futureDatesDisabled
+                    ? isDateInFuture(fullDate)
+                    : false;
+
+                  return (
+                    <DateTableCell
+                      $isCurrentMonth={isCurrentMonth}
+                      $isDisabled={isDisabled}
+                      $isSelected={isSelected}
+                      $isToday={isCurrentDate}
+                      key={dayKey}
+                      onClick={() => {
+                        if (isDisabled) {
+                          return false;
+                        }
+                        setSelectedDate(fullDate);
+                        closeDatepicker();
+                      }}
+                    >
+                      {date}
+                    </DateTableCell>
+                  );
+                })}
+              </tr>
+            );
+          })}
+        </tbody>
+      </DateTable>
+    </DatePickerContainer>
+  );
+};
+
+export interface DatePickerProps {
+  date?: Date;
```

Comment:
Given this will be very close to an input, why not keeping the API consistent with what we usually find on inputs?

instead of date => value
instead of onSelectDate => onChange ?

Reply:
>why not keeping the API consistent with what we usually find on inputs

Because it's not a generic input, it's a date picker. `onChange` makes sense for an event that is fired on each keystroke that changes the value of the input. `onSelect` makes sense when selecting an item from a select list. When a date is selected, `onSelectDate` is fired, because that's the action the user has taken.